app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharactersView.kt

```diff
@@ -0,0 +1,56 @@
+package com.puzzlebench.clean_marvel_kotlin.presentation.mvp.view
+
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import com.puzzlebench.clean_marvel_kotlin.R
+import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
+import com.puzzlebench.clean_marvel_kotlin.presentation.CharacterDetailFragment
+import com.puzzlebench.clean_marvel_kotlin.presentation.CharacterActivity
+import com.puzzlebench.clean_marvel_kotlin.presentation.adapter.CharacterAdapter
+import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
+import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.contracts.CharactersContract
+import kotlinx.android.synthetic.main.activity_main.recycleView
+import kotlinx.android.synthetic.main.activity_main.progressBar
+import java.lang.ref.WeakReference
+
+
+class CharactersView(activity: CharacterActivity): CharactersContract.View {
+    private val activityRef = WeakReference(activity)
+    var adapter = CharacterAdapter(emptyList()) { character ->
```

Comment:
this var, needs to be var and public?

Reply:
Done! You were right, now adapter is val and private!